draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,76 +60,333 @@ collector). Thus it is up to each deployment of DAP to securely implement its
+     I-D.draft-ietf-ppm-dap-02. */
+    QueryConfig query_config;
+
+    /* The maximum number of times a batch of reports may be queried by
+    the Collector. */
+    uint16 max_batch_lifetime;
+
+    /* Time up to which Clients are allowed to upload to this task. See
+    https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/304. Defined
+    in I-D.draft-ietf-ppm-dap-02. */
+    Time task_expiration;
+
+    /* A codepoint defined in I-D.draft-irtf-cfrg-vdaf-03 or reserved for private
+    use. */
+    VdafType vdaf_type;
+
+    /* Additional parameters relevant for the vdaf_type. */
+    opaque vdaf_config<1..2^16-1>;
+} TaskConfig;
+
+struct {
+    QueryType query_type;    // Defined in I-D.draft-ietf-ppm-dap-02
+    Duration time_precision; // Defined in I-D.draft-ietf-ppm-dap-02
+    uint32 min_batch_size;
+    select (query_type) {
+        case time-interval: Empty;
+        case fixed-size: uint32 max_batch_size;
+    }
+} QueryConfig;
 ~~~
 
+The purpose of `TaskConfig` is to define all parameters that are necessary
+for configuring an aggregator. It includes all the fields to be
+associated with a task (see task configuration in
+{{?DAP=I-D.draft-ietf-ppm-dap-01}}.). Besides, `TaskConfig` also includes fields
+that useful for configuring a task in-band:
+
+* An opaque `task_info` that is specific to a task. For e.g. this can be a
+string describing the purpose of this task. It can also be the DAP task ID
+chosen by task author.
+
+* An opaque `vdaf_config` that contains any VDAF specific parameters for the
+chosen `vdaf_type`. The aggregators MUST pass `vdaf_config` to VDAF initialiser,
+based on the chosen `vdaf_type`.
+
+The codepoints for standardized (V)DAFs are listed below:
+
+~~~
+/* Codepoint for each standardized VDAF. Defined in
+ I-D.draft-irtf-cfrg-vdaf-02 */
+enum {
+    Prio3Aes128Count(0x00000000),
+    Prio3Aes128Sum(0x00000001),
+    Prio3Aes128Histogram(0x00000002),
+    Poplar1Aes128(0x00001000),
+    (2^32-1)
+} VdafType;
+~~~
+
+`VdafConfig` is not specified in this document, instead it should be defined by
+each VDAF implementation, for example, the simplest VDAF config for Prio3 can
+be defined as:
+
+~~~
+struct {
+    select (vdaf_type) {
+        case Prio3Aes128Count: Empty;
+        case Prio3Aes128Sum: uint8 bits;
+        case Prio3Aes128Histogram: uint32 buckets;
+    }
+} VdafConfig;
+~~~
+
+> OPEN ISSUE: Should DP parameters be defined as a different "dimension" to
+> VDAF, given that it's likely various DP mechanisms can be applied to any VDAF.
+
+The definition of Time, Duration, Url, QueryType follow those in
+{{?DAP=I-D.draft-ietf-ppm-dap-01}}.
+
+Note that `TaskConfig` does not encode all of the parameters required for the
+aggregator to run a task. In particular, parameters that should not be known to
+clients, like the VDAF verification key, the collector HPKE configuration, and
+whatever assets are required for HTTP request authentication are still
+established out-of-band.
```

Comment:
I think this extension should spell out derivation of the VDAF verification key, as this needs to be unique per task. Feel free to leave as an OPEN ISSUE in this PR.